File: tests/modscan/src/net/neoforged/fml/common/EventBusSubscriber.java
package net.neoforged.fml.common;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import net.neoforged.api.distmarker.Dist;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface EventBusSubscriber {
    Dist[] value() default { Dist.CLIENT, Dist.DEDICATED_SERVER };

    String modid() default "";
}

File: library/src/libmatti/net/neoforged/fml/common/EventBusSubscriber.c
// Port of net.neoforged.fml.common.EventBusSubscriber.

#include "libmatti/net/neoforged/fml/common/EventBusSubscriber.h"

// Java: value() default { Dist.CLIENT, Dist.DEDICATED_SERVER }
static const LIBMATTI_DIST_Dist DEFAULT_SIDES[2] = {LIBMATTI_DIST_CLIENT, LIBMATTI_DIST_DEDICATED_SERVER};

LIBMATTI_FML_EventBusSubscriber LIBMATTI_FML_EventBusSubscriber_Defaults(void)
{
    LIBMATTI_FML_EventBusSubscriber subscriber = {DEFAULT_SIDES, 2, ""};
    return subscriber;
}
